molo/profiles/models.py

```diff
@@ -80,10 +123,39 @@ class UserProfile(models.Model):
         null=True)
 
     mobile_number = PhoneNumberField(blank=True, null=True, unique=False)
+    security_question_answers = models.ManyToManyField(
+        SecurityQuestion,
+        through="SecurityAnswer"
+    )
 
 
 @receiver(post_save, sender=User)
 def user_profile_handler(sender, instance, created, **kwargs):
     if created:
         profile = UserProfile(user=instance)
         profile.save()
+
+
+class SecurityAnswer(models.Model):
+    user = models.ForeignKey(UserProfile)
+    question = models.ForeignKey(SecurityQuestion)
```

Comment:
Valid point. I agree. However, taking that approach assumes prior knowledge - the user is identified before being asked the questions.

That would then mean that this process has to be broken into two steps:
1) Firstly, we check whether the user exists by checking against the username.
2) Having found the user, we then obtain the particular security questions.

That in itself weakens the security a bit in my opinion.
What are your thoughts in light of this?